cheetah/accelerator.py

```diff
@@ -327,6 +335,431 @@ def defining_features(self) -> list[str]:
     def __repr__(self) -> str:
         return f"{self.__class__.__name__}(length={repr(self.length)})"
 
+class SpaceChargeKick(Element):
+    """
+    Applies the effect of space charge over a length `length`, on the **momentum**
+    (i.e. divergence and energy spread) of the beam.
```

Comment:
This first part uncommon/confusing from my experience. I would make this a thin element (L=0) and apply it instantaneously multiple times; slice the element to apply it in.

The rest below is great (nice doc string!! :sparkling_heart: )

Reply:
Yes, I agree that calling this `length` is confusing.

Note that, in the current PR, the slicing is done by hand by the user. For instance, in the CI test,
```
Segment(
    elements=[
        Drift(length=L/6),
       SpaceChargeKick(length=L/3),
       Drift(length=L/3),
       SpaceChargeKick(length=L/3),
       Drift(length=L/3),
       SpaceChargeKick(length=L/3),
       Drift(length=L/6)
  ]
)
```
corresponds to breaking the element `Drift(length=L)` into 3 slices.

But as discussed in the PR and in issues, this should not be the final interface and their should be a follow-up PR to introduce a more user-friendly interface.

Reply:
I agree, for this PR that is ok and we can finalize this part in a follow-up as discussed.